File: src/Usuario.java

public class Usuario {
  private PerfilJugador perfil;

  // Constructors
  public Usuario() {
  }

  public Usuario(String nombre, String email, String telefono, String passwordHash) {
    this.nombre = nombre;
    this.email = email;
    this.telefono = telefono;
    this.passwordHash = passwordHash;
  }

}

File: src/Usuario.ts
import { PerfilJugador } from "./PerfilJugador";

export class Usuario {
  id?: number;
  nombre?: string;
  email?: string;
  telefono?: string;
  passwordHash?: string;
  perfil?: PerfilJugador;

  constructor(nombre?: string, email?: string, telefono?: string, passwordHash?: string) {
    if (nombre && email && telefono && passwordHash) {
      this.nombre = nombre;
      this.email = email;
      this.telefono = telefono;
      this.passwordHash = passwordHash;
    }
  }

}